server/validation/validation.go

```diff
@@ -43,5 +46,30 @@ func ValidateObjectsRelations(ctx context.Context, backend storage.TypeDefinitio
 	if !ok {
 		return nil, &tuple.RelationNotFoundError{Relation: t.GetRelation(), TypeName: ns.GetType(), TupleKey: t}
 	}
+
 	return userset, nil
 }
+
+// ValidateUserIfItsAUserset returns an error if the t.User is a userset and its type/relation don't exist in the modelID
+func ValidateUserIfItsAUserset(ctx context.Context, backend storage.TypeDefinitionReadBackend, store string, modelID string, t *openfgapb.TupleKey) error {
```

Comment:
Don't take a `TypeDefinitionReadBackend`. Let's pass either the model OR the typesystem.

Reply:
I could do this refactor, so that both `ValidateObjectsRelations` and `ValidateUserIfItsAUserset` take in a typesystem, but aren't you going to change that code anyway after https://github.com/openfga/openfga/pull/341 ?

Reply:
Yeah, I guess so.